Data Manager Documentation Update (#951)
CitrineInformatics/citrine-python: docs/source/data_entry.rst

    dataset.register(ProcessSpec(...))

Note that registration must be performed within the scope of a dataset: the dataset into which the objects are being written.

> The data model object collections that are defined with the team scope (such as `team.process_specs`) are read-only and will throw an error if their register method is called.

Reviewer: ```suggestion
The data model object collections that are defined with the team scope (such as `team.process_specs`) are read-only.
Attempts to register, update, etc. via those collections will throw an error.
```
